iml-api/src/graphql.rs

```diff
@@ -392,6 +384,56 @@ impl QueryRoot {
 
         Ok(snapshots)
     }
+    /// List all snapshot intervals
+    async fn list_snapshot_intervals(
+        context: &Context,
+    ) -> juniper::FieldResult<Vec<SnapshotInterval>> {
+        let xs: Vec<SnapshotInterval> = sqlx::query!("SELECT * FROM snapshot_interval")
+            .fetch(&context.pg_pool)
+            .map_ok(|x| SnapshotInterval {
+                id: x.id,
+                filesystem_name: x.filesystem_name,
+                use_barrier: x.use_barrier,
+                interval: x.interval.into(),
+                last_run: x.last_run.map(|x| DateTime::from(x)),
+            })
+            .try_collect()
+            .await?;
+
+        Ok(xs)
+    }
+    /// List all snapshot retention policies
+    async fn list_snapshot_retention_policies(
+        context: &Context,
+    ) -> juniper::FieldResult<Vec<SnapshotRetention>> {
+        let xs: Vec<SnapshotRetention> = sqlx::query!(
+            r#"
+                SELECT
+                    id,
```

Comment:
I didn't think about this earlier but I guess this could be:
```
SELECT * FROM snapshot_retention
```

Reply:
no, because delete_unit is being cast.